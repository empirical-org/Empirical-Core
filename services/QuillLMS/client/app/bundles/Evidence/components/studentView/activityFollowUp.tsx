import * as React from 'react'

import PostActivitySlide from '../activitySlides/postActivitySlide'
import ThankYouSlide from '../activitySlides/thankYouSlide'
import ActivitySurvey from '../activitySlides/activitySurvey'

const ActivityFollowUp = ({ activity, dispatch, responses, sessionID, saveActivitySurveyResponse, }) => {
  const [showActivitySurvey, setShowActivitySurvey] = React.useState(false)
  const [submittedActivitySurvey, setSubmittedActivitySurvey] = React.useState(false)

  function onClickNext() { setShowActivitySurvey(true) }

  if (submittedActivitySurvey) {
    return <ThankYouSlide />
  }

  if (showActivitySurvey) {
    return (
      <ActivitySurvey
        activity={activity}
        dispatch={dispatch}
        saveActivitySurveyResponse={saveActivitySurveyResponse}
        sessionID={sessionID}
        setSubmittedActivitySurvey={setSubmittedActivitySurvey}
      />
    )
  }
  return <PostActivitySlide handleClick={onClickNext} prompts={activity && activity.prompts} responses={responses} />
}

export default ActivityFollowUp
