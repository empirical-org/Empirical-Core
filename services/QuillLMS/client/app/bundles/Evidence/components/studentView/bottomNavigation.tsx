import * as React from 'react'

import { bigCheckIcon, } from '../../../Shared/index'

const MINIMUM_STUDENT_HIGHLIGHT_COUNT = 2

const ReadAndHighlightTracker = ({
  scrolledToEndOfPassage,
  studentHighlights,
  handleClickDoneHighlighting,
  handleReadTheDirectionsButtonClick,
  showReadTheDirectionsButton
}) => {
  const minimumMet = studentHighlights.length >= MINIMUM_STUDENT_HIGHLIGHT_COUNT
  let doneButton = <button className="quill-button contained primary large focus-on-light disabled" onClick={handleClickDoneHighlighting} type="button">Done</button>
  if (scrolledToEndOfPassage && minimumMet) {
    doneButton = <button className="quill-button contained primary large focus-on-light" onClick={handleClickDoneHighlighting} type="button">Done</button>
  }
  if(showReadTheDirectionsButton) {
    return(
      <div className="read-and-highlight-tracker bottom-navigation read-instructions">
        <button aria-label="Next" className="quill-button contained primary large focus-on-light" onClick={handleReadTheDirectionsButtonClick} type="button">Got it</button>
      </div>
    )
  }
  return (
    <div className="read-and-highlight-tracker bottom-navigation">
      {doneButton}
      <div aria-hidden={true} className="read-and-highlight-steps">
        <div className="read-and-highlight-step">
          {scrolledToEndOfPassage ? <img alt={bigCheckIcon.alt} className="check-icon" src={bigCheckIcon.src} /> : <div className="incomplete-indicator" />}
          <span>Read the entire text</span>
        </div>
        <div className="read-and-highlight-step">
          {minimumMet ? <img alt={bigCheckIcon.alt} className="check-icon" src={bigCheckIcon.src} /> : <div className="incomplete-indicator" />}
          <span>Highlight two sentences</span>
        </div>
      </div>
    </div>
  )
}

const BottomNavigation = ({
  scrolledToEndOfPassage,
  studentHighlights,
  handleClickDoneHighlighting,
  handleReadTheDirectionsButtonClick,
  doneHighlighting,
  handleDoneReadingClick,
  hasStartedPromptSteps,
  hasStartedReadPassageStep,
  handleStartPromptStepsClick,
  handleStartReadingPassageClick,
  inReflection,
  showReadTheDirectionsButton,
  showStepsSummary,
  toggleShowStepsSummary
}) => {

  if (!hasStartedReadPassageStep) {
    return (
      <div className="bottom-navigation">
        <button className="quill-button outlined secondary large focus-on-dark" onClick={handleStartReadingPassageClick} type="button">Start</button>
      </div>
    )
  }

  if (hasStartedReadPassageStep && !doneHighlighting) {
    return (
      <ReadAndHighlightTracker
        handleClickDoneHighlighting={handleClickDoneHighlighting}
        handleReadTheDirectionsButtonClick={handleReadTheDirectionsButtonClick}
        scrolledToEndOfPassage={scrolledToEndOfPassage}
        showReadTheDirectionsButton={showReadTheDirectionsButton}
        studentHighlights={studentHighlights}
      />
    )
  }

  if (inReflection) {
    return (
      <div className="bottom-navigation">
        <button className="quill-button contained primary large focus-on-light" onClick={handleDoneReadingClick} type="button">Next</button>
      </div>
    )
  }

  if (!hasStartedPromptSteps) {
    return (
      <div className="bottom-navigation">
        <button className="quill-button outlined secondary large focus-on-dark" onClick={handleStartPromptStepsClick} type="button">Next</button>
      </div>
    )
  }

  if (showStepsSummary) {
    return (
      <div className="bottom-navigation">
        <button className="quill-button outlined secondary large focus-on-dark" onClick={toggleShowStepsSummary} type="button">Next</button>
      </div>
    )
  }
}

export default BottomNavigation
