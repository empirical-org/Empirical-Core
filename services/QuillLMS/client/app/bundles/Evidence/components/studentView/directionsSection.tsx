import * as React from 'react'

import { DEFAULT_HIGHLIGHT_PROMPT, } from '../../../Shared/utils/constants'
import { informationIcon } from '../../../Shared/index'

const READ_PASSAGE_STEP = 1
const promptDirections = ['Use information from the text to finish the sentence.', 'Put the information in your own words.']
const highlightDirections = ['First, read the highlighting task below.', 'Then, read the text and highlight sentences to complete the task.']

const renderDirections = (directionsTextArray: string[]) => (
  <section className="directions-section">
    <h3>Directions</h3>
    <ul>
      {directionsTextArray.map((text: string) => (
        <li>{text}</li>
      ))}
    </ul>
  </section>
)

const essentialKnowledgeSection = ({ essential_knowledge_text }) => {
  if(!essential_knowledge_text || essential_knowledge_text === '<br/>') { return }

  return (
    <div className="essential-knowledge-section">
      <div className="essential-knowledge-header">
        <img alt={informationIcon.alt} src={informationIcon.src} />
        <h3>Building Essential Knowledge</h3>
      </div>
      <div className="essential-knowledge" dangerouslySetInnerHTML={{ __html: essential_knowledge_text }} />
    </div>
  )
}

const DirectionsSection = ({ className, passage, inReflection, activeStep, }) => {
  const uniquePartOfHighlightPrompt = passage.highlight_prompt ? passage.highlight_prompt.replace(DEFAULT_HIGHLIGHT_PROMPT, '') : ''
  if (inReflection) {
    return (
      <div className={className}>
        <section className="reflection-section">
          <h3>Directions</h3>
          <p>Great! Now take a moment to reflect on the sentences you highlighted. The ideas you highlighted may be helpful as you complete the writing prompts in the next section.</p>
        </section>
      </div>
    )
  }

  if (activeStep > READ_PASSAGE_STEP) {
    return renderDirections(promptDirections)
  }

  return (
    <div className={className}>
      {renderDirections(highlightDirections)}
      <section className="task-section">
        <h3>Task</h3>
        <p>{DEFAULT_HIGHLIGHT_PROMPT}<u>{uniquePartOfHighlightPrompt}</u></p>
      </section>
      {essentialKnowledgeSection(passage)}
    </div>
  )
}

export default DirectionsSection
