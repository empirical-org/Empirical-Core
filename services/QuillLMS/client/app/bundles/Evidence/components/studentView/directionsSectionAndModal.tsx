import * as React from 'react'

import { DEFAULT_HIGHLIGHT_PROMPT, } from '../../../Shared/utils/constants'

const READ_PASSAGE_STEP = 1

export const ReadTheDirectionsModal = ({ closeReadTheDirectionsModal, }) => (<section className="read-the-directions-modal">
  <p>Read the directions carefully before moving onto reading and highlighting.</p>
  <button className="quill-button primary contained large focus-on-light" onClick={closeReadTheDirectionsModal} type="button">Got it</button>
</section>)


const DirectionsSectionAndModal = ({ className, closeReadTheDirectionsModal, passage, showReadTheDirectionsModal, inReflection, activeStep, }) => {
  const uniquePartOfHighlightPrompt = passage.highlight_prompt ? passage.highlight_prompt.replace(DEFAULT_HIGHLIGHT_PROMPT, '') : ''
  if (inReflection) {
    return (
      <div className={className}>
        <section className="reflection-section">
          <h3>Directions</h3>
          <p>Great! <u>Now take a moment to reflect on the sentences you highlighted.</u></p>
        </section>
      </div>
)
  }

  if (activeStep > READ_PASSAGE_STEP) {
    return (
      <div className={className}>
        <section className="directions-section">
          <h3>Directions</h3>
          <ul>
            <li>Use information from the text to finish the sentence.</li>
            <li>Put the information in your own words.</li>
          </ul>
        </section>
      </div>
)
  }

  return (
    <div className={className}>
      {showReadTheDirectionsModal ? <ReadTheDirectionsModal closeReadTheDirectionsModal={closeReadTheDirectionsModal} /> : <span />}
      <section className="directions-section">
        <h3>Directions</h3>
        <p>{DEFAULT_HIGHLIGHT_PROMPT}<u>{uniquePartOfHighlightPrompt}</u></p>
      </section>
    </div>
)
}

export default DirectionsSectionAndModal
