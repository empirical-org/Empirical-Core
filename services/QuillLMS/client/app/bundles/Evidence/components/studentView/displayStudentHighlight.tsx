import * as React from 'react'

import { removeIcon } from '../../../Shared/index'

const DisplayStudentHighlight = ({ studentHighlight, removeHighlight, inReflection, }) => {
  function handleClickRemove() { removeHighlight(studentHighlight)}

  return (
    <div className="display-student-highlight">
      <span>{studentHighlight}</span>
      {inReflection ? <span /> : <button aria-label={`Remove highlight from the following sentence: ${studentHighlight}`} className="interactive-wrapper focus-on-light" onClick={handleClickRemove} type="button"><img alt={removeIcon.alt} src={removeIcon.src} /></button>}
    </div>
  )
}

export default DisplayStudentHighlight
