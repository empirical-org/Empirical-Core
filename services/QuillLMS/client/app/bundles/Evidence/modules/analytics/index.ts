import { Event } from './event_definitions';
import Events from './events';


class SegmentAnalytics {
  analytics(): object {
    return window.analytics
  }

  track(event: Event, properties?: object): boolean {
    try {
      // Make sure that the event reference is one that's defined
      if (!event) {
        throw new Error('The event referenced does not exist.');
      }

      // Validate that required properties are present
      this.validateEvent(event, properties);

      // Check to make sure that we have access to the analytics global
      if (!this.analytics()) {
        throw new Error(`Error sending event '${event.name}'.  SegmentAnalytics was instantiated before an instance of window.analytics could be found to connect to.`);
      }
    } catch(e) {
      this.reportError(e);
      return false;
    }

    const eventProperties = Object.assign(this.formatCustomProperties(properties), this.getDefaultProperties());

    this.analytics().track(event.name, eventProperties);
    return true;
  }

  validateEvent (event: Event, properties?: object): void {
    if (properties === undefined) {
      properties = {};
    }
    if (event.requiredProperties) {
      let passedEventProperties = Object.keys(properties);
      event.requiredProperties.forEach((p) => {
        if (passedEventProperties.indexOf(p) == -1) {
          throw new Error(`Can not track event "${event.name}" without required property "${p}".`);
        }
      });
    }
  }

  formatCustomProperties(properties: object): object {
    if (typeof properties != 'object') {
      properties = {};
    }
    return Object.keys(properties).reduce((accumulator, key) => {
      let customKeyName = `custom_${key}`;
      accumulator[customKeyName] = properties[key];
      return accumulator;
    }, {});
  }

  getDefaultProperties(): object {
    return {
      path: `${window.location.pathname}${window.location.search}${window.location.hash}`,
      referrer: document.referrer,
    };
  }

  reportError(e: Error): void {
    // placeholder for actual error reporting
    // to do - do something with this error
  }
}

const segmentInstance = new SegmentAnalytics();

export {
  segmentInstance as SegmentAnalytics,
  Events,
};
