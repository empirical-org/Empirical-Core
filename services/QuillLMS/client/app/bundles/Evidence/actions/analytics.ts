import * as request from 'request';

import { Event } from '../modules/analytics/event_definitions'
import { SegmentAnalytics } from '../modules/analytics'

export const TrackAnalyticsEvent = (event: Event, params: Object) => {
  return (dispatch: Function) => {
    SegmentAnalytics.track(event, params);
  }
}
