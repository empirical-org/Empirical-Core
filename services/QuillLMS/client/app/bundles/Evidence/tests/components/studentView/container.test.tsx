import * as React from 'react'
import { shallow, mount, } from 'enzyme';
import toJson from 'enzyme-to-json';

import { activityOne } from './data'

const csrfToken = 'mocked-csrf-token';
document.head.innerHTML = `<meta name="csrf-token" content="${csrfToken}">`;

const mockParse = () => ({ uid: activityOne.activity_id })
jest.mock('query-string', () => ({
  default: {
    parse: mockParse
  }
}))

const mockTrackAnalyticsEvent = jest.fn()
jest.mock('../../../actions/analytics', () => ({
  TrackAnalyticsEvent: mockTrackAnalyticsEvent
}))

const mockGetActivity = jest.fn()
jest.mock('../../../actions/activities', () => ({
  getActivity: mockGetActivity
}))

jest.mock('string-strip-html', () => ({
  default: jest.fn(() => {
    return {};
  })
}))

import { StudentViewContainer } from '../../../components/studentView/container'
import LoadingSpinner from '../../../components/shared/loadingSpinner'
import PromptStep from '../../../components/studentView/promptStep'
import { Events } from '../../../modules/analytics'

const dispatch = () => {}

const activitiesReducer = { hasReceivedData: true, currentActivity: activityOne}
const sessionReducer = { submittedResponses: [], sessionID: 'MockSessionID', hasReceivedData: true }

const wrapper = mount(<StudentViewContainer
  activities={activitiesReducer}
  dispatch={dispatch}
  location={{ search: "?uid=1" }}
  session={sessionReducer}
/>)

// TODO: update tests to fix activity and prompt IDs for mockTrackAnalyticsEvent calls

// TODO: add tests for transformMarkTags function

describe('StudentViewContainer component', () => {
  describe('when the activity has loaded', () => {

    const scrollToStepMock = jest.fn()
    wrapper.instance().scrollToStep = scrollToStepMock
    wrapper.setState({ explanationSlidesCompleted: true })

    it('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should track a COMPREHENSION_ACTIVITY_STARTED event', () => {
      expect(mockGetActivity).toHaveBeenCalledWith(sessionReducer.sessionID, activityOne.activity_id)
    })

    describe('when the first step has been completed', () => {
      wrapper.setState({ completedSteps: [1], activeStep: 2, hasStartedReadPassageStep: true, hasStartedPromptSteps: true, })
      it('should render a promptStep for each prompt in the activity', () => {
        expect(wrapper.find(PromptStep).length).toBe(activityOne.prompts.length)
      })

      it('should render prompts in correct order', () => {
        expect(wrapper.find(PromptStep).at(0).props().prompt.conjunction).toEqual('because')
        expect(wrapper.find(PromptStep).at(1).props().prompt.conjunction).toEqual('but')
        expect(wrapper.find(PromptStep).at(2).props().prompt.conjunction).toEqual('so')
      });

    })

    describe('when complete step is called', () => {
      describe('when the next step has not yet been completed and is not more than the total number of steps', () => {
        beforeAll(() => {
          wrapper.setState({ completedSteps: [1] })
          wrapper.instance().completeStep(2)
        })

        it('sets the next chronological step as the active step', () => {
          expect(wrapper.state('activeStep')).toBe(3)
          expect(wrapper.state('completedSteps')).toEqual([1, 2])
        })

        it('should also track a COMPREHENSION_PROMPT_COMPLETED event', () => {
          expect(mockTrackAnalyticsEvent).toHaveBeenNthCalledWith(1, Events.COMPREHENSION_PROMPT_COMPLETED, {
            activityID: undefined,
            sessionID: sessionReducer.sessionID,
            promptID: activityOne.prompts[0].id
          })
        })

        it('should also track a COMPREHENSION_PROMPT_STARTED event', () => {
          expect(mockTrackAnalyticsEvent).toHaveBeenNthCalledWith(2, Events.COMPREHENSION_PROMPT_STARTED, {
            activityID: undefined,
            sessionID: sessionReducer.sessionID,
            promptID: activityOne.prompts[1].id
          })
        })
      })

      describe('when the next step has already been completed', () => {
        it('sets the first uncompleted step as the active step', () => {
          wrapper.setState({ completedSteps: [2] })
          wrapper.instance().completeStep(1)
          expect(wrapper.state('activeStep')).toBe(3)
          expect(wrapper.state('completedSteps').sort()).toEqual([1, 2])
        })
      })

      describe('when the next step is more than the total number of steps', () => {
        it('sets the first uncompleted step to be the active step', () => {
          wrapper.setState({ completedSteps: [1, 3] })
          wrapper.instance().completeStep(4)
          expect(wrapper.state('activeStep')).toBe(2)
          expect(wrapper.state('completedSteps')).toEqual([1, 3, 4])
        })
      })

      describe('when all steps are completed', () => {
        beforeAll(() => {
          mockTrackAnalyticsEvent.mockClear()
          wrapper.setState({ completedSteps: [1, 2, 3] })
          wrapper.instance().completeStep(4)
        })

        it('adds the last step to the completed steps array', () => {
          expect(wrapper.state('completedSteps')).toEqual([1, 2, 3, 4])
        })

        it('should also track a COMPREHENSION_ACTIVITY_COMPLETED event', () => {
          expect(mockTrackAnalyticsEvent).toHaveBeenNthCalledWith(2, Events.COMPREHENSION_ACTIVITY_COMPLETED, {
            activityID: activityOne.activity_id,
            sessionID: sessionReducer.sessionID
          })
        })
      })
    })

    describe('when the user clicks on a stepLink', () => {
      const scrollToStepOnMobileMock = jest.fn()
      wrapper.instance().scrollToStepOnMobile = scrollToStepOnMobileMock

      describe('when the read passage step has been completed', () => {
        beforeAll(() => {
          mockTrackAnalyticsEvent.mockClear()
          wrapper.setState({ completedSteps: [1], activeStep: 2, hasStartedReadPassageStep: true, hasStartedPromptSteps: true, activityIsComplete: false, explanationSlidesCompleted: true })
          wrapper.find('.step-links').find('.step-link').last().simulate('click')
        })

        it('activates that step', () => {
          expect(wrapper.state('activeStep')).toBe(4)
          expect(scrollToStepOnMobileMock).toHaveBeenCalled()
        })

        it('should also track a COMPREHENSION_PROMPT_STARTED event', () => {
          const lastPromptIndex = activityOne.prompts.length - 1
          expect(mockTrackAnalyticsEvent).toHaveBeenCalledWith(Events.COMPREHENSION_PROMPT_STARTED, {
            activityID: undefined,
            sessionID: sessionReducer.sessionID,
            promptID: activityOne.prompts[lastPromptIndex].id
          })
        })
      })
    })

  })

  describe('when the activity has not loaded', () => {
    const activitiesReducer = { hasReceivedData: false }
    const sessionReducer = { submittedResponses: {} }
    const wrapper = shallow(<StudentViewContainer
      activities={activitiesReducer}
      dispatch={dispatch}
      location={{ search: "?uid=1" }}
      session={sessionReducer}
    />)

    it('renders a loading spinner', () => {
      expect(wrapper.find(LoadingSpinner).length).toBe(1)
    })
  })

})
