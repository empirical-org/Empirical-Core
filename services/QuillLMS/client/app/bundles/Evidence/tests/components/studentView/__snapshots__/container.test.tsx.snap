// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentViewContainer component renders 1`] = `
<StudentViewContainer
  activities={
    Object {
      "currentActivity": Object {
        "activity_id": 1,
        "passages": Array [
          Object {
            "highlight_prompt": "As you read, highlight two sentences that explain why some people think voting should be required.",
            "text": "Crispin Morales is a poor Bolivian factory worker who works 12-hour days. He works every day but Sunday. The last presidential election fell on a Saturday. Although he was given the day off as the law required, Crispin was unable to get to a polling station to vote. After failing to vote, Crispin was not allowed to withdraw his salary from his bank for over two months. This is because those who do not vote are punished in Bolivia given that voting there is compulsory, a civic duty required by law. Not all countries with compulsory voting enforce this law.

If a citizen of Australia or Brazil provides a good reason for not voting (e.g. being in a hospital) it is accepted. In fact, many nations have dropped mandatory voting altogether. However, this can have a negative effect. When Venezuela, for example, discontinued mandatory voting in 1993, it saw a 30% decline in attendance at the polls.

Low voter turnout is a common problem in countries without compulsory voting. This problem might be explained by voter apathy or lack of enthusiasm around elections. Of 222 million eligible voters, only 125 million cast their votes in the 2012 U.S. presidential election. By contrast, over 750 million votes were cast during season 10 of American Idol. Some people find it discouraging that voting for American Idol seems to generate more enthusiasm than voting in national elections. While it's important to note that American Idol viewers as young as 13 years old can vote multiple times during multiple episodes, these numbers may indicate that some citizens feel that their vote counts more in television contests than in political contests. Or perhaps theyâ€™re just not interested in politics.

Mandatory voting may seem like a great way to encourage voter participation, but governments can sometimes misuse it. For example, in some dictatorships, citizens are forced to vote, but the same person is elected every time because the government does not allow political competition. In March 2014 in North Korea, 99% of the country cast votes for Kim Jong Un, the only candidate listed on the ballot.

But how can democracies have representative governments unless all or most of their citizens vote? Some say that compulsory voting ensures that the elected government represents the majority of the population. But those against compulsory voting say that it is a violation of personal liberties and choices. Others say that voting should be seen as a civic right and not a civic duty.",
          },
        ],
        "prompts": Array [
          Object {
            "conjunction": "because",
            "max_attempts": 5,
            "max_attempts_feedback": "Nice effort! You made some strong revisions. Here is an example of a strong response. What is similar or different about your response? 

 Governments should make voting compulsory because otherwise not everyone will vote.",
            "prompt_id": 1,
            "text": "Governments should make voting compulsory because",
          },
          Object {
            "conjunction": "but",
            "max_attempts": 5,
            "max_attempts_feedback": "Nice effort! You made some strong revisions. Here is an example of a strong response. What is similar or different about your response? 

 Governments should make voting compulsory, but with exceptions for people who have good reasons for not being able to vote.",
            "prompt_id": 2,
            "text": "Governments should make voting compulsory, but",
          },
          Object {
            "conjunction": "so",
            "max_attempts": 5,
            "max_attempts_feedback": "Nice effort! You made some strong revisions. Here is an example of a strong response. What is similar or different about your response? 

 Governments should make voting compulsory so that the elected government represents the majority of the population.",
            "prompt_id": 3,
            "text": "Governments should make voting compulsory so",
          },
        ],
        "title": "Should Voting be Required?",
      },
      "hasReceivedData": true,
    }
  }
  dispatch={[Function]}
  location={
    Object {
      "search": "?uid=1",
    }
  }
  session={
    Object {
      "activeStep": 1,
      "explanationSlidesCompleted": false,
      "hasReceivedData": true,
      "sessionID": "MockSessionID",
      "submittedResponses": Array [],
    }
  }
>
  <WelcomeSlide
    onHandleClick={[Function]}
  >
    <div
      className="explanation-slide-container"
      id="welcome-slide-container"
    >
      <section
        id="information-section"
      >
        <p
          className="subtext"
        >
          Hi there!
        </p>
        <section
          id="header-container"
        >
          <p
            id="header"
          >
            Welcome to Quill Reading for Evidence
          </p>
        </section>
        <section
          id="instructions-container"
        >
          <div
            className="instruction-container"
          >
            <div
              className="step-number"
            >
              1
            </div>
            <p
              className="step-text"
            >
              Read and highlight text
            </p>
            <img
              alt="A sample article about volcanoes showing a sentence highlighted with the cursor over the highlighted sentence"
              className="step-image"
              src="undefined/images/evidence/article-highlighted.svg"
            />
          </div>
          <div
            className="instruction-container"
          >
            <div
              className="step-number"
            >
              2
            </div>
            <p
              className="step-text"
            >
              Write sentences using what you read
            </p>
            <img
              alt="A sample question asking a student to fill in the rest of a sentence"
              className="step-image"
              src="undefined/images/evidence/sentence-prompt.svg"
            />
          </div>
          <div
            className="instruction-container"
          >
            <div
              className="step-number"
            >
              3
            </div>
            <p
              className="step-text"
            >
              Revise based on feedback
            </p>
            <img
              alt="A sample feedback asking a student to add more detail to their sentence"
              className="step-image"
              src="undefined/images/evidence/sentence-feedback.svg"
            />
          </div>
        </section>
      </section>
      <Footer
        buttonText="Next"
        onHandleClick={[Function]}
        step={1}
      >
        <section
          id="button-container"
        >
          <span />
          <button
            className="quill-button large secondary outlined focus-on-dark"
            onClick={[Function]}
            type="submit"
          >
            Next
          </button>
          <div
            id="step-indictator-container"
          >
            <div
              className="step-indicator active"
            />
            <div
              className="step-indicator "
            />
            <div
              className="step-indicator "
            />
            <div
              className="step-indicator "
            />
          </div>
        </section>
      </Footer>
    </div>
  </WelcomeSlide>
</StudentViewContainer>
`;
